Objects/bytearrayobject.c

```diff
@@ -2601,8 +2704,13 @@ static PyObject *
 bytearray_sizeof_impl(PyByteArrayObject *self)
 /*[clinic end generated code: output=738abdd17951c427 input=e27320fd98a4bc5a]*/
 {
-    size_t res = _PyObject_SIZE(Py_TYPE(self));
-    res += (size_t)FT_ATOMIC_LOAD_SSIZE_RELAXED(self->ob_alloc) * sizeof(char);
+    Py_ssize_t res = _PyObject_SIZE(Py_TYPE(self));
+    Py_ssize_t alloc = FT_ATOMIC_LOAD_SSIZE_RELAXED(self->ob_alloc) * sizeof(char);
+    if (alloc > 0) {
+        res += sizeof(PyBytesObject);
```

Comment:
Not sure the CPython internals way to do this. The storage must always be exactly a `PyBytesObject` currently so added size; can use `_PyObject_SIZE` or similar if that makes more sense.